qancode.py

```diff
@@ -0,0 +1,2366 @@
+            print('{}Outputting file {}{}'.format(
+                bcolors.FAIL, path_name, bcolors.ENDC))
+            diff = cv2.addWeighted(image_one, 0.2, difference, 1, 0)
+            all_viz = np.concatenate([image_one, diff, image_two], axis=1)
+            cv2.imwrite(os.path.join(directory, path_name), all_viz)
+        else:
+            self.diff_found = False
+            # cv2.imwrite(os.path.join(directory, 'match_one.png'), image_one)
+            # cv2.imwrite(os.path.join(directory, 'match_two.png'), image_two)
+            print('{}MATCH{}'.format(bcolors.OKBLUE, bcolors.ENDC))
+        return (self.diff_found, path_name)
+
+    def compare_data(self):
+        print('\nComparing screenshots between URLs.')
+        print('As user: {}'.format(self.user))
+        print('Browser: {}'.format(self.browser))
+        print('First URL: {}'.format(self.prod_url))
+        print('Second URL: {}'.format(self.rc_url))
+        print('Item type: {}'.format(self.item_type))
+        print('Click path: {}'.format(
+            None if self.click_path is None else self.click_path.__name__))
+        result = self.compute_image_difference()
+        print('Distance metric: {}'.format(self.diff_distance_metric))
+        return result
+
+
+################
+# Click paths. #
+################
+
+# --- Genome browsers. ---
+
+class OpenUCSCGenomeBrowser(object):
+    """
+    Defines clicks required to open UCSC trackhub from Experiment page for
+    given assembly.
+    """
+
+    def __init__(self, driver, assembly):
+        self.driver = driver
+        self.assembly = assembly
+        self.perform_action()
+
+    def perform_action(self):
+        current_window = self.driver.current_window_handle
+        time.sleep(1)
+        for y in self.driver.find_elements_by_tag_name(ExperimentPage.all_buttons_tag_name):
+            try:
+                if y.text == 'Visualize':
+                    y.click()
+                    self.driver.wait.until(EC.element_to_be_clickable(
+                        (By.CLASS_NAME, VisualizeModal.modal_class)))
+                    break
+            except:
+                pass
+        modal = self.driver.wait.until(
+            EC.element_to_be_clickable((By.CLASS_NAME, VisualizeModal.modal_class)))
+        UCSC_links = modal.find_elements_by_partial_link_text(
+            VisualizeModal.UCSC_link_partial_link_text)
+        for link in UCSC_links:
+            if self.assembly in link.get_attribute("href"):
+                print('Opening genome browser')
+                link.click()
+                break
+        time.sleep(1)
+        self.driver.switch_to_window([h for h in self.driver.window_handles
+                                      if h != current_window][0])
+        self.driver.wait.until(EC.element_to_be_clickable(
+            (By.ID, UCSCGenomeBrowser.zoom_one_id)))
+        time.sleep(3)
+
+
+class OpenUCSCGenomeBrowserGRCh38(object):
```

Comment:
these should definitely NOT be sub classes.   They are just instances of the parent class with a specific parameter